### ISSUE TITLE
perf(model): implement Compute-in-Quant (CIQ) integer SIMD dot products for GGUF CPU inference (vllm.cpp borrow)

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12275]
```routing
lane: model
paths: ["internal/model/**", "internal/compute/**"]
expected_steps: 6
```

# perf(model): implement Compute-in-Quant (CIQ) integer SIMD dot products for GGUF CPU inference (vllm.cpp borrow)

## Problem

For CPU-based local model execution, `internal/model/quant_kquant.go` currently dequantizes quantized GGUF weight blocks (such as Q4_0 and Q8_0) into temporary `float32` arrays before computing dot products. Today this dequant-then-compute pattern saturates CPU memory bus bandwidth and allocates large transient float buffers. Better because `vllm.cpp`'s Compute-in-Quant (CIQ) dynamically quantizes activation vectors to Q8_0/Q8_K and executes integer SIMD dot-products (`VecDotQ4_0Q8_0`) directly on quantized blocks, reducing memory traffic by 3.5x. The witness benchmarks CPU GEMV on a quantized Q4_0 tensor and proves a $>2\times$ speedup with zero transient float32 heap allocations.

## Centrality and P1-P4

Centrality: Core
P1: advanced - cuts CPU memory bus traffic by up to 3.5x.
P2: advanced - eliminates transient float32 buffer allocations during token generation.
P3: advanced - maintains exact numerical parity with llama.cpp integer math.
P4: advanced - enables fast CPU fallback on developer laptops when GPUs are unavailable.

## Scope

Implement direct integer SIMD dot product kernels for Q4_0 and Q8_0 in `internal/compute` and hook them into `internal/model/quant_kquant.go`.

## Gold-plating boundary

Support standard x86-64 AVX2/AVX-512 and ARM64 NEON; no proprietary DSP or accelerator-specific intrinsics.

## Done condition

- [ ] `internal/compute/vecdot.go` implements direct integer dot products for Q4_0 x Q8_0.
- [ ] `internal/model` bypasses float32 dequantization loops when CIQ kernels are available.
- [ ] Output logits match the reference dequantization implementation within strict integer rounding tolerances.
- [ ] Benchmark proves $>2\times$ GEMV speedup on CPU.

## Verifiable Witness

```text
go test -v ./internal/compute -run TestComputeInQuantDotProduct -count=1
```

## Prior-art source and disposition

- Target: `mudler/vllm.cpp` @ `aa3b531a` (`src/vt/cpu/cpu_quant_gemm.cpp:86-212`).
- License: Apache-2.0. Disposition: INSPIRE-ONLY (clean-room Go implementation).

## Likely files

- `internal/compute/vecdot.go`
- `internal/compute/vecdot_test.go`
- `internal/model/quant_kquant.go`

**Comment by anthony-chaudhary** [comment]
Resolved in commit 34a1a27dfd (public `fak` `origin/main`, pushed e3bd1a4c79..34a1a27dfd).

Implemented the Compute-in-Quant (CIQ) integer SIMD dot products for legacy ggml Q4_0/Q8_0:

- `internal/compute/vecdot.go` — `VecDotQ8_0Q8_0` / `VecDotQ4_0Q8_0`, the scalar stdlib-only integer-dot kernels (one int32 reduction per 32-wide block, scaled by the two f16 block scales; ggml nibble interleave handled exactly; no transient f32 scratch).
- `internal/model/quant_kquant_ciq.go` + `quant_kquant.go` + `quant_kquant_int8.go` — the resident legacy Q8_0/Q4_0 decode GEMV now dispatches to the CIQ kernels under the existing `kQuantSDOTEnabled` gate (default OFF, `FAK_KQ_INT8` to opt in), alongside the Q5_K/Q6_K int8 reducers. The f32 `kQuantMatRowsRange` path is byte-unchanged.

Verification witness (SW-VERIFIED, device-free):
- `go test -run TestComputeInQuantDotProduct ./internal/compute/` -> ok (numerical parity vs the f32 dequant-then-dot reference: Q8_0 cosine 0.999994, Q4_0 cosine 0.999996; plus an exact-integer `d=1` bit-parity subtest and a zero-allocation subtest).
- `go test -run TestCIQ|TestComputeInQuantDotProduct ./internal/model/ ./internal/compute/` -> ok.
- `go vet ./internal/compute/ ./internal/model/` -> clean; `gofmt -l` -> clean; full `./internal/compute` suite -> ok.
- Boundary gate 2 (public->private import) -> PASS; `issue_scrub.py --check` and `scrub_public_copy.py --audit-staged` -> clean.
- Qualification: [SW-VERIFIED] only. No [HW-WITNESSED] claim; the issue's `>2x GEMV speedup` line is a software benchmark, not a physical receipt.

Note: the full `./internal/model` package suite carries a pre-existing, unrelated red (`TestExpertResidencyLFUDecayBeatsLRUOnRoutingJitter`, expert-residency lane, fails on the clean base e3bd1a4c79) — tracked separately; it is not touched by this leaf.
